daemon/daemon_windows.go

```diff
@@ -36,12 +36,14 @@ func checkKernel() error {
 // adaptContainerSettings is called during container creation to modify any
 // settings necessary in the HostConfig structure.
 func (daemon *Daemon) adaptContainerSettings(hostConfig *runconfig.HostConfig, adjustCPUShares bool) {
-	if hostConfig.CPUShares < 0 {
-		logrus.Warnf("Changing requested CPUShares of %d to minimum allowed of %d", hostConfig.CPUShares, windowsMinCPUShares)
-		hostConfig.CPUShares = windowsMinCPUShares
-	} else if hostConfig.CPUShares > windowsMaxCPUShares {
-		logrus.Warnf("Changing requested CPUShares of %d to maximum allowed of %d", hostConfig.CPUShares, windowsMaxCPUShares)
-		hostConfig.CPUShares = windowsMaxCPUShares
+	if hostConfig != nil {
```

Comment:
Just for consistency with the unix version can you do it like this:
https://github.com/docker/docker/blob/master/daemon/daemon_unix.go#L95